addons/mail/static/src/components/discuss/discuss.js

```diff
@@ -68,9 +68,22 @@ class Discuss extends Component {
 
     patched() {
         this.trigger('o-update-control-panel');
-        if (this.discuss.thread) {
+        if (this.discuss.thread &&
+            this.discuss.thread.isPinned &&
```

Comment:
This seems misused, hints are only for propagating visual clues to components.

All the logic you do here is models logic only, this should be done inside models only (ideally with compute methods, maybe in handlers, at worse with updateAfter).